content/tutorials/build-apps/build-a-desktop-wallet-in-javascript.md

```diff
@@ -0,0 +1,1460 @@
+
+4. Next, modify the `view/template.html` file by adding placeholders to the `<body>` section:
+
+```html
+<body>
+    <!-- Step 2 code additions - start -->
+    <h3>Build a XRPL Wallet - Part 2/8</h3>
+    <b>Latest validated ledger stats</b><br />
+    Ledger Index: <strong id="ledger-index"></strong><br />
+    Ledger Hash: <strong id="ledger-hash"></strong><br />
+    Close Time: <strong id="ledger-close-time"></strong><br />
+    <!-- Step 2 code additions - end -->
+</body>
+```
+
+5. Modify the `view/renderer.js` file to handle the new placeholders. Note that the renamed preloader function is now reflected in `window.electronAPI.onUpdateLedgerData`:
+
+```javascript
+const ledgerIndexEl = document.getElementById('ledger-index')
+const ledgerHashEl = document.getElementById('ledger-hash')
+const ledgerCloseTimeEl = document.getElementById('ledger-close-time')
+
+window.electronAPI.onUpdateLedgerData((_event, value) => {
+    ledgerIndexEl.innerText = value.ledger_index
+    ledgerHashEl.innerText = value.ledger_hash
+    ledgerCloseTimeEl.innerText = value.ledger_time
+})
+```
+
+This should make our application listen to regular updates of the ledger and display them in the frontend.
+
+Now run the application with the following command:
+
+```console
+npm run start
+```
+
+This time the application should be the same as in the last step, with the difference that the latest ledger index value gets updated roughly very 3-5 seconds.
+
+To run the reference application found in `content/_code-samples/build-a-wallet/desktop-js` for this step, run:
+
+```console
+npm run async
+```
+
+### 3. Display an Account
+
+**Full code for this step:**
+[`library/3_helpers.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/3_helper.js).
+[`3-account/index.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/3_account.js),
+[`3-account/view/preload.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/3_preload.js),
+[`3-account/view/renderer.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/3_renderer.js),
+[`3-account/view/template.html`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/3_account.html).
+
+We now have a permanent connection to the XRPL and some code to bring the delivered data to life on our screen, it's time to add some "wallet" functionality by managing an individual account. 
+
+We will ask the user for address of the account to monitor by using a HTML dialog element. We will furthermore refactor the application by encapsulating some functionality in a library. After finishing this step the application should look like this:
+
+![Screenshot: Step 3, show account information](img/javascript-wallet-3.png)
```

Comment:
It seems like in this screenshot, and the next several screenshots we have `XRP Reserved` as a field, but it is never populated. I think either the field should not be a part of these earlier steps if there's no logic to populate it, or there should be logic to populate it.

Reply:
As it is never really used, I will remove it.

Reply:
Removed